zstackctl/zstackctl/license_manifest.py

```
@@ -0,0 +1,273 @@
+STATUS_TAMPERED = 'tampered'
+STATUS_SIGNATURE_INVALID = 'signature_invalid'
+STATUS_MISSING_MANIFEST = 'missing_manifest'
+STATUS_ERROR = 'error'
+
+# Verification modes, in decreasing order of trust.
+MODE_NATIVE = 'native'              # authoritative: pinned-fingerprint + signature + hashes
+MODE_HASH_ONLY = 'hash_only'        # degraded: file hashes only, signature/anchor unchecked
+
+
+def _verify_with_helper(helper_path, manifest_path, pubkey_path, root):
+    proc = subprocess.Popen([helper_path, '--manifest', manifest_path,
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)